src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1900 @@ interface MPSMatrixMultiplication {
+
+		[Export ("initWithSource:integerScaleFactorX:integerScaleFactorY:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint integerScaleFactorX, nuint integerScaleFactorY);
+
+		[Export ("scaleFactorX")]
+		double ScaleFactorX { get; }
+
+		[Export ("scaleFactorY")]
+		double ScaleFactorY { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNUpsamplingBilinearNode")]
+	interface MPSCnnUpsamplingBilinearNode
+	{
+		[Static]
+		[Export ("nodeWithSource:integerScaleFactorX:integerScaleFactorY:")]
+		MPSCnnUpsamplingBilinearNode Create (MPSNnImageNode sourceNode, nuint integerScaleFactorX, nuint integerScaleFactorY);
+
+		[Export ("initWithSource:integerScaleFactorX:integerScaleFactorY:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint integerScaleFactorX, nuint integerScaleFactorY);
+
+		[Export ("scaleFactorX")]
+		double ScaleFactorX { get; }
+
+		[Export ("scaleFactorY")]
+		double ScaleFactorY { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel), Name = "MPSNNGraph")]
+	interface MPSNnGraph : NSCopying, NSSecureCoding
+	{
+		[Export ("initWithDevice:resultImage:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSNnImageNode resultImage);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("sourceImageHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] SourceImageHandles { get; }
+
+		[NullAllowed, Export ("sourceStateHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] SourceStateHandles { get; }
+
+		[NullAllowed, Export ("intermediateImageHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] IntermediateImageHandles { get; }
+
+		[NullAllowed, Export ("resultStateHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] ResultStateHandles { get; }
+
+		[NullAllowed, Export ("resultHandle")]
+		IMPSHandle ResultHandle { get; }
+
+		[Export ("outputStateIsTemporary")]
+		bool OutputStateIsTemporary { get; set; }
+
+		[Export ("destinationImageAllocator", ArgumentSemantic.Retain)]
+		IMPSImageAllocator DestinationImageAllocator { get; set; }
+
+		[Export ("encodeToCommandBuffer:sourceImages:sourceStates:intermediateImages:destinationStates:")]
+		MPSImage Encode (IMTLCommandBuffer commandBuffer, MPSImage[] sourceImages, [NullAllowed] MPSState[] sourceStates, [NullAllowed] NSMutableArray<MPSImage> intermediateImages, [NullAllowed] NSMutableArray<MPSState> destinationStates);
+
+		[Export ("encodeToCommandBuffer:sourceImages:")]
+		MPSImage Encode (IMTLCommandBuffer commandBuffer, MPSImage[] sourceImages);
+
+		[Async, Export ("executeAsyncWithSourceImages:completionHandler:")]
+		MPSImage Execute (MPSImage[] sourceImages, Action<MPSImage, NSError> handler);
+	}
+
+	interface IMPSHandle { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
```

Comment:
That `[Model]` is really not fine, it's meant as an interface (so a protocol is enough), a concrete type won't help (but ill confuse developers)